src/charm_state.py

```diff
@@ -771,9 +774,11 @@ def from_charm(cls, charm: CharmBase) -> list["SSHDebugConnection"]:
                 )
                 continue
             ssh_debug_connections.append(
+                # pydantic allows string to be passed as IPvAnyAddress and as int,
+                # mypy complains about it
                 SSHDebugConnection(
-                    host=host,
-                    port=port,
+                    host=host,  # type: ignore
```

Comment:
Why not casting instead?

Reply:
This was from a cherry picked commit, that does sound better, thanks!

Reply:
The values are supposed to be automatically parsed from `str` / `int` to `IPv4Address` and `int` type by Pydantic automatically - I think it might be better not to cast it before Pydantic handles it.